server/conn.go

```diff
@@ -604,7 +604,12 @@ func (cc *clientConn) addMetrics(cmd byte, startTime time.Time, err error) {
 	} else {
 		metrics.QueryTotalCounter.WithLabelValues(label, "OK").Inc()
 	}
-	metrics.QueryDurationHistogram.WithLabelValues(metrics.LblGeneral).Observe(time.Since(startTime).Seconds())
+	stmtType := cc.ctx.GetSessionVars().StmtCtx.StmtType
+	sqlType := metrics.LblGeneral
+	if stmtType != "" {
+		sqlType = stmtType
+	}
+	metrics.QueryDurationHistogram.WithLabelValues(sqlType).Observe(time.Since(startTime).Seconds())
```

Comment:
Why do `QueryDurationHistogram` and `QueryTotalCounter` use the different `label`？

Reply:
I think it's a "historical reasons", old code choose COMMAND as count label and none label for the duration, 

To add duration label, IMO stmtType(like insert/select) is more useful than command(like COM_QUERY).

why don't change QueryTotalCounter is maybe sometime we need that, and we also have another `StmtNodeCounter` to cover "count by stmtType" although has some litte different.

so after this we can see:

- count of command ->  QueryTotalCounter
- count of stmt -> StmtNodeCounter
- duration of query -> QueryDurationHistogram and optional switch to stmtType view

:D